tsconfig.json

```
@@ -2,24 +2,31 @@
 	"compilerOptions": {
 		"allowJs": true,
 		"checkJs": true,
-		"noEmit": true,
-		"module": "es2015",
+		"emitDeclarationOnly": true,
+		"declaration": true,
+		"module": "esnext",
 		"moduleResolution": "node",
-		"target": "es5",
-		"baseUrl": "packages",
-		"paths": {
-			"@tannin/*": [ "*/index.js" ],
-			"tannin": [ "tannin/index.js" ]
-		}
+		"target": "esnext",
```

Aside from the `import`/`export` module semantics (handled by Rollup), all code in this project is assumed to run in ES5, so I don't think this change would be desirable. I would want to flag errors when using code which doesn't work in an ES5 environment.

---

If you're not using TypeScript for compilation (we're not here, just for generating the declarations), then TypeScript isn't a good tool to guarantee `es5`.

Arrow functions`() => {}` are acceptable to with an es5 target, for example.

At the same time, an `es5` target _will_ determine the `lib`:

> Note: If --lib is not specified a default list of libraries are injected. The default libraries injected are:
> ► For `--target ES5`: `DOM,ES5,ScriptHost`
> ► For `--target ES6`: `DOM,ES6,DOM.Iterable,ScriptHost`

`lib` effectively controls the feature set available, so e.g. `WeakMap;` would become an error. That can be specified apart from `target`, for example if this should be Node.js compatible and we'd like to exclude the DOM APIs.

`es5` is probably a good default, I'll apply that change.

```suggestion
		"target": "es5",
```

---

That's good to highlight. It wasn't entirely clear to me what effect `target` actually has here. Off the top of my head, I think ESLint might be more effective in my original goal, via parser configuration. Though I specifically recall needing to configure this as ES2015, due to how ESLint refuses to allow a combination of `sourceType: 'module'` and `ecmaVersion: 5`.

https://github.com/aduth/tannin/blob/cd1c7447843df7751c4abd1b92aee03fe56bfba4/.eslintrc.json#L4-L7